### PR TITLE
Remove PeripheralMarker

### PR DESCRIPTION
**Pull request by bugadani**
This PR untangles the dma module from peripherals that don't use/support it, by removing PeripheralMarker.

### CODE REVIEW

**Comment by MabezDev** [comment]
Could we drop everything but the last commit here, so we can unblock the P4 PR?

**Review by MabezDev — APPROVED** [review]
LGTM, just curious about hiding from the docs

File: esp-hal/src/twai/mod.rs [hunk 1]
```diff
@@ -1472,10 +1471,13 @@ where
 }
 
 /// TWAI peripheral instance.
-pub trait Instance: Peripheral<P = Self> + PeripheralMarker + Into<AnyTwai> + 'static {
+pub trait Instance: Peripheral<P = Self> + Into<AnyTwai> + 'static {
     /// The identifier number for this TWAI instance.
     fn number(&self) -> usize;
 
+    #[doc(hidden)]
```
[the remark this answers is not in the document]
Comment (bugadani) [reply esp-hal/src/twai/mod.rs:1478]: This whole thing will eventually be removed anyway

**Review by playfulFence — APPROVED** [review]
Thanks, `gdma` was already cfg-d out from `P4` PR!

**Pull request merged**